iree/turbine/kernel/wave/index_sequence_analysis.py

```diff
@@ -80,25 +81,42 @@ def has_strided_access(node: fx.Node) -> bool:
             for dim in custom.index
         }
 
-        max_stride = int(max(simplified_index[dim].stride for dim in simplified_index))
         shape = get_vector_shape(hw_constraint, custom.register_type.symbolic_shape)
         elements_per_thread = subs_idxc(custom.elements_per_thread)
+        max_stride_dim, max_stride = max(
+            [(dim, seq.stride) for dim, seq in simplified_index.items()],
+            key=lambda item: item[1],
+        )
         with custom.graph.inserting_before(operator):
             for i in range(elements_per_thread):
+                # Non-contiguous access patterns can have varying offsets. We
+                # handle that here.
+                gpr_offset = [
+                    expr
+                    for expr in simplified_index[max_stride_dim].start.args
+                    if expr.has(GPR_NUM)
+                ]
+                if not gpr_offset:
+                    gpr_offset = i
+                else:
+                    gpr_offset = sympy.Add(*gpr_offset).subs({GPR_NUM: i})
```

Comment:
IIUC, we are trying to sub in `GPR_NUM: i` on simplified_index[max_stride_dim].start, BUT without the lane information. Wouldn't be more accurate with the lane info i,e (`4 * floor(lane / 32)`)? if that is not the case, can you explain why not? 🤔

Reply:
You are right. We need both the lane offset and gpr offset. The lane offset is added later when computing write.index below.